scripts/gen_hw_fixture_table.sh

```diff
@@ -0,0 +1,92 @@
+#! /bin/bash
+#
+# Copyright (c) 2024 Linutronix GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if [ x"$1" == "x" ]
+then
+    echo "usage: $0 <hardware-fixture-table>"
+    exit 1
+fi
+
+TOPDIR=$(git rev-parse --show-toplevel)
+
+find $TOPDIR -name testcase.yaml -exec awk '/fixture:/{print $2}' {} \; | \
+    sort -u | \
+    awk -voutfile="$1" '
```

Comment:
When I try the script with, for example, one fixture removed from the doc file to imitate a new fixture appears, the script adds it with no description. It is a bit confusing. 
May it be a link to the testcase.yaml file(s) where the fixture appears, or even a cross-reference column here to point what test suites need the fixture ?

```
~/zephyrproject/zephyr$ git diff
diff --git a/doc/develop/test/twister/hardware-fixtures.rst b/doc/develop/test/twister/hardware-fixtures.rst
index 06325ed4798..d57b969417e 100644
--- a/doc/develop/test/twister/hardware-fixtures.rst
+++ b/doc/develop/test/twister/hardware-fixtures.rst
@@ -28,8 +28,6 @@ Hardware fixture table
 +----------------------------------+--------------------------------------------------------------------------------------------+
 | i2c_bus_short                    | The DUT has two I2C busses connected together and the I2C controller are DUAL role capable |
 +----------------------------------+--------------------------------------------------------------------------------------------+
-| i2c_loopback                     | The DUT has an I2C master controller connected to an I2C slave controller                  |
-+----------------------------------+--------------------------------------------------------------------------------------------+
 | i2c_ram                          | The DUT has a MB85 FeRAM connected to an I2C bus                                           |
 +----------------------------------+--------------------------------------------------------------------------------------------+
 | maxim_ds3231                     | The DUT is equipped with a Maxim DS3231 chip                                               |

~/zephyrproject/zephyr$ ./scripts/gen_hw_fixture_table.sh ./doc/develop/test/twister/hardware-fixtures.rst

~/zephyrproject/zephyr$ git diff
diff --git a/doc/develop/test/twister/hardware-fixtures.rst b/doc/develop/test/twister/hardware-fixtures.rst
index 06325ed4798..b2927355613 100644
--- a/doc/develop/test/twister/hardware-fixtures.rst
+++ b/doc/develop/test/twister/hardware-fixtures.rst
@@ -28,7 +28,7 @@ Hardware fixture table
 +----------------------------------+--------------------------------------------------------------------------------------------+
 | i2c_bus_short                    | The DUT has two I2C busses connected together and the I2C controller are DUAL role capable |
 +----------------------------------+--------------------------------------------------------------------------------------------+
-| i2c_loopback                     | The DUT has an I2C master controller connected to an I2C slave controller                  |
+| i2c_loopback                     |                                                                                            |
 +----------------------------------+--------------------------------------------------------------------------------------------+
 | i2c_ram                          | The DUT has a MB85 FeRAM connected to an I2C bus                                           |
 +----------------------------------+--------------------------------------------------------------------------------------------+
```